includes/class-llms-awards-query.php

```diff
@@ -0,0 +1,525 @@
+		 *
+		 * @since [version]
+		 *
+		 * @param object  $object     The retrieved object.
+		 * @param string  $award_type The award type for the query, either "achievements" or "certificates".
+		 * @param WP_Post $post       The original object.
+		 */
+		return apply_filters( 'llms_awards_query_post_type', $object, $this->type, $post );
+
+	}
+
+	/**
+	 * Retrieve the post type for the query's award type.
+	 *
+	 * @since [version]
+	 *
+	 * @return string
+	 */
+	private function get_post_type() {
+
+		$post_type = '';
+		if ( 'achievements' === $this->type ) {
+			$post_type = 'llms_my_achievement';
+		} elseif ( 'certificates' === $this->type ) {
+			$post_type = 'llms_my_certificate';
+		}
+
+		/**
+		 * Filters the post type for the query's award type.
+		 *
+		 * @since [version]
+		 *
+		 * @param string $post_type  A WP_Post_Type name.
+		 * @param string $award_type The award type for the query, either "achievements" or "certificates".
+		 */
+		return apply_filters( 'llms_awards_query_post_type', $post_type, $this->type );
+
+	}
+
+	/**
+	 * Retrieve the results for the given query.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $output Determine the return type. Either "OBJECTS" to return `LLMS_User_Achievement` or `LLMS_User_Certificate` objects
+	 *                       or `POSTS` to return `WP_Post` objects.
+	 * @return LLMS_User_Achievement[]|LLMS_User_Certificate[]|WP_Post[] Array of objects.
+	 */
+	public function get_results( $output = 'OBJECTS' ) {
+
+		$posts = $this->query->posts;
+
+		if ( 'POSTS' === $output ) {
+			return $posts;
+		}
+
+		return array_map( array( $this, 'get_object' ), $posts );
+
+	}
+
+	/**
+	 * Retrieve the meta key name for the template used to generate the awarded post.
+	 *
+	 * @since [version]
+	 *
+	 * @return string
+	 */
+	private function get_template_meta_key() {
+
+		$meta_key = '';
+		if ( 'achievements' === $this->type ) {
+			$meta_key = '_llms_achievement_template';
+		} elseif ( 'certificates' === $this->type ) {
+			$meta_key = '_llms_certificate_template';
+		}
+
+		/**
+		 * Filters the meta key name for the query's award type.
+		 *
+		 * @since [version]
+		 *
+		 * @param string $meta_key   The meta key name.
+		 * @param string $award_type The award type for the query, either "achievements" or "certificates".
+		 */
+		return apply_filters( 'llms_awards_query_meta_key', $meta_key, $this->type );
+
+	}
+
+	/**
+	 * Retrieve the total number of available result pages for the given query.
+	 *
+	 * If `no_found_rows` is supplied to the initial query this will return `0`
+	 * which may not reflect the actual number of result pages.
+	 *
+	 * @since [version]
+	 *
+	 * @return int
+	 */
+	public function get_total_pages() {
+		return $this->query->max_num_pages;
+	}
+
+	/**
+	 * Determine if any results were found for the query.
+	 *
+	 * @since [version]
+	 *
+	 * @return boolean
+	 */
+	public function has_results() {
+		return count( $this->query->posts ) > 0;
+	}
+
```

Comment:
Maybe we can add this useful method too:
https://github.com/gocodebox/lifterlms/blob/e9e27855088754b27b2fc2010fe69260327eb5a2/includes/abstracts/abstract.llms.database.query.php#L268

Reply:
I know, I'm giving you feedbacks while I use it <3